test/junit/scala/tools/nsc/backend/jvm/IndyLambdaTest.scala

```diff
@@ -0,0 +1,54 @@
+package scala.tools.nsc.backend.jvm
+
+import org.junit.Assert._
+import org.junit.{Assert, Test}
+
+import scala.tools.asm.{Handle, Opcodes}
+import scala.tools.asm.tree.InvokeDynamicInsnNode
+import scala.tools.nsc.backend.jvm.AsmUtils._
+import scala.tools.nsc.backend.jvm.CodeGenTools._
+import scala.tools.testing.ClearAfterClass
+import scala.collection.JavaConverters._
+
+object IndyLambdaTest extends ClearAfterClass.Clearable {
+  var compiler = newCompiler(extraArgs = "-Ybackend:GenBCode")
+
+  def clear(): Unit = {
+    compiler = null
+  }
+}
+
+class IndyLambdaTest {
```

Comment:
You need some more boilerplate: the class needs to extend `ClearAfterClass`, and `ClearAfterClass.stateToClear` needs to be set. Maybe this can be improved..
```
class IndyLambdaTest extends ClearAfterClass {
  ClearAfterClass.stateToClear = IndyLambdaTest
  ...
```